src/requirements/data/majors/ling.ts

```diff
@@ -0,0 +1,95 @@
+import { Course, CollegeOrMajorRequirement } from '../../types';
+
+import {
+  courseIsFWS,
+  ifCodeMatch,
+  courseMatchesCodeOptions,
+  includesWithSubRequirements,
+} from '../checkers-common';
+
+const lingRequirements: readonly CollegeOrMajorRequirement[] = [
+  {
+    name: 'Foundation Courses',
+    description: 'LING 1101, LING 3302, LING 3303, and LING 3314.',
+    source: 'https://linguistics.cornell.edu/undergraduate#major-requirements:',
+    checker: includesWithSubRequirements(
+      ['LING 1101'],
+      ['LING 3302'],
+      ['LING 3303'],
+      ['LING 3314']
+    ),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1, 1, 1, 1],
+  },
+  {
+    name: 'Additional Courses: 3000 or above',
+    description: 'At least 2 additional courses must be at the 3000 level or above.',
+    source: 'https://linguistics.cornell.edu/undergraduate#major-requirements:',
+    checker: [
+      (course: Course): boolean =>
+        ifCodeMatch(course.subject, 'LING') &&
+        !(ifCodeMatch(course.catalogNbr, '1***') || ifCodeMatch(course.catalogNbr, '2***')) &&
+        !courseMatchesCodeOptions(course, ['LING 1101', 'LING 3302', 'LING 3303', 'LING 3314']) &&
+        !courseIsFWS(course),
+    ],
+    fulfilledBy: 'courses',
+    perSlotMinCount: [2],
+  },
+  {
+    name: 'Additional Courses: 2000 or above',
+    description: 'At least 3 additional courses must be 2000 level or above.',
+    source: 'https://linguistics.cornell.edu/undergraduate#major-requirements:',
+    checker: [
+      (course: Course): boolean =>
+        ifCodeMatch(course.subject, 'LING') &&
+        !ifCodeMatch(course.catalogNbr, '1***') &&
+        !courseMatchesCodeOptions(course, ['LING 1101', 'LING 3302', 'LING 3303', 'LING 3314']) &&
+        !courseIsFWS(course),
+    ],
+    fulfilledBy: 'courses',
+    perSlotMinCount: [3],
+  },
+  {
+    name: 'Additional Courses: 1000 or above',
+    description: 'At most 1 additional course can be 1000 level or above.',
+    source: 'https://linguistics.cornell.edu/undergraduate#major-requirements:',
+    checker: [
+      (course: Course): boolean =>
+        ifCodeMatch(course.subject, 'LING') &&
+        !courseMatchesCodeOptions(course, ['LING 1101', 'LING 3302', 'LING 3303', 'LING 3314']) &&
+        !courseIsFWS(course),
+    ],
+    fulfilledBy: 'courses',
+    perSlotMinCount: [1],
+  },
+  {
+    name: 'Additional Courses: 18 credits',
+    description: 'Additional courses must each be 3 credits or more.',
+    source: 'https://linguistics.cornell.edu/undergraduate#major-requirements:',
+    checker: [
+      (course: Course): boolean =>
+        ifCodeMatch(course.subject, 'LING') &&
+        !courseMatchesCodeOptions(course, ['LING 1101', 'LING 3302', 'LING 3303', 'LING 3314']) &&
+        !courseIsFWS(course),
+    ],
+    allowCourseDoubleCounting: true,
+    fulfilledBy: 'credits',
+    perSlotMinCount: [18],
+  },
```

Comment:
> Is there any way we can restrict courses to only display ones that are 3 (or more) credits? Right now I have an 18 credit checker for the additional courses, which is not technically correct, as 2 credit courses could be used to fulfill this requirement when they can't in reality.

I think you could check the credits by looking at `unitsMinimum` or `unitsMaximum`? Checking `unitsMaximum` may be better than checking `unitsMinimum`. A helper function that checks that maximum credits a course could possibly be would be nice to reuse and have in `src/requirements/data/checkers-common.ts`

https://github.com/cornell-dti/course-plan/blob/c002a9ac0585d1aad76a6bc024a9038b2eb25877/src/requirements/types.ts#L1-L18